### PR TITLE
[test framework] Adding local pilot

### PR DESCRIPTION
**Pull request by nmittler**

### CODE REVIEW

**Comment by nmittler** [comment]
FYI, I've created a separate PR against master to add the required configuration to pilot bootstrap: #6075

**Comment by nmittler** [comment]
@ozevren @kimikowang PTAL

File: pkg/test/local/environment.go [hunk 1]
```diff
@@ -91,8 +100,11 @@ func (e *Environment) GetMixerOrFail(t testing.TB) environment.DeployedMixer {
 
 // GetPilot returns a deployed Pilot instance in the environment.
 func (e *Environment) GetPilot() (environment.DeployedPilot, error) {
-	// TODO
-	panic("Not yet implemented")
+	p, err := e.get(dependency.Pilot)
```
[the remark this answers is not in the document]
Comment (ozevren) [reply pkg/test/local/environment.go]: The dependency tracker logic should take care of duplication issues:
https://github.com/istio/istio/blob/0ef0df56d4af572d1fe9fb94eccf5dd0c6c679c3/pkg/test/impl/driver/instance.go#L206
https://github.com/istio/istio/blob/0ef0df56d4af572d1fe9fb94eccf5dd0c6c679c3/pkg/test/internal/tracker.go#L33

File: pkg/test/local/environment.go [hunk 2]
```diff
@@ -18,10 +18,17 @@ import (
...
+	envoyv1 "istio.io/istio/pilot/pkg/proxy/envoy/v1"
 	"istio.io/istio/pkg/test/dependency"
 	"istio.io/istio/pkg/test/environment"
 	"istio.io/istio/pkg/test/fakes/policy"
 	"istio.io/istio/pkg/test/internal"
+	"istio.io/istio/pkg/test/local/pilot"
+)
+
+const (
+	namespace = "istio-system"
```
Comment (ozevren) [inline pkg/test/local/environment.go]: This is not used yet, but can you add a TODO to consume this from the environment member?
https://github.com/istio/istio/blob/0ef0df56d4af572d1fe9fb94eccf5dd0c6c679c3/pkg/test/cluster/environment.go#L34

**Pull request merged**